src/subclass/cell_renderer.rs

```diff
@@ -0,0 +1,658 @@
+            flags,
+        )
+    }
+
+    fn start_editing<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        event: Option<&gdk::Event>,
+        widget: &P,
+        path: &str,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    ) -> Option<CellEditable> {
+        self.parent_start_editing(
+            renderer,
+            event,
+            widget,
+            path,
+            background_area,
+            cell_area,
+            flags,
+        )
+    }
+}
+
+pub trait CellRendererImplExt {
+    fn parent_get_request_mode(&self, renderer: &CellRenderer) -> SizeRequestMode;
+    fn parent_get_preferred_width<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+    ) -> (i32, i32);
+    fn parent_get_preferred_width_for_height<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        height: i32,
+    ) -> (i32, i32);
+    fn parent_get_preferred_height<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+    ) -> (i32, i32);
+    fn parent_get_preferred_height_for_width<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        width: i32,
+    ) -> (i32, i32);
+    fn parent_get_aligned_area<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        widget: &P,
+        flags: CellRendererState,
+        cell_area: &gdk::Rectangle,
+    ) -> gdk::Rectangle;
+    fn parent_render<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        cr: &cairo::Context,
+        widget: &P,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    );
+    fn parent_activate<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        event: &gdk::Event,
+        widget: &P,
+        path: &str,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    ) -> bool;
+    fn parent_start_editing<P: IsA<Widget>>(
+        &self,
+        renderer: &CellRenderer,
+        event: Option<&gdk::Event>,
+        widget: &P,
+        path: &str,
+        background_area: &gdk::Rectangle,
+        cell_area: &gdk::Rectangle,
+        flags: CellRendererState,
+    ) -> Option<CellEditable>;
+}
+
+impl<T: CellRendererImpl + ObjectImpl> CellRendererImplExt for T {
+    fn parent_get_request_mode(&self, renderer: &CellRenderer) -> SizeRequestMode {
+        unsafe {
+            let data = self.get_type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut gtk_sys::GtkCellRendererClass;
+            if let Some(f) = (*parent_class).get_request_mode {
+                from_glib(f(renderer.to_glib_none().0))
+            } else {
+                SizeRequestMode::__Unknown(0)
+            }
```

Comment:
As the C code unconditionally uses the vfunc in `gtk_cell_renderer_get_request_mode()` we should panic if it does not exist

Reply:
I wasn't sure here so I took the safe route. Changing it.